builtin/logical/aws/path_static_roles.go

```diff
@@ -202,14 +205,38 @@ func (b *backend) pathStaticRolesDelete(ctx context.Context, req *logical.Reques
 }
 
 func (b *backend) validateRoleName(name string) error {
+	if name == "" {
+		return errors.New("empty role name attribute given")
+	}
 	return nil
 }
 
 func (b *backend) validateIAMUserExists(ctx context.Context, req *logical.Request, username string) error {
+	c, err := b.clientIAM(ctx, req.Storage)
+	if err != nil {
+		return fmt.Errorf("unable to validate username '%s': %w", username, err)
+	}
+
+	// we don't really care about the content of the result, just that it's not an error
+	out, err := c.GetUser(&iam.GetUserInput{
+		UserName: aws.String(username),
+	})
+	if err != nil || out.User == nil || *out.User.UserName != username {
+		return fmt.Errorf("unable to validate username '%s': %w", username, err)
+	}
 	return nil
 }
 
+const (
+	minAllowableSeconds = 1
+	maxAllowableSeconds = -1
+)
+
 func (b *backend) validateRotationPeriod(period time.Duration) error {
+	secs := period.Seconds()
+	if secs < minAllowableSeconds || (maxAllowableSeconds != -1 && secs > maxAllowableSeconds) {
```

Comment:
Should `maxAllowableSeconds` not be a constant? Otherwise I don't see how it could be anything other than -1.

Reply:
Yeah I think it's either that or having it be configurable via config - I'm not sure how much validation we do on other static role ttl/rotation periods.

(either way it wouldn't be a constant, you're right, for this mid-implementation PR I mostly just wanted something there calling it out)

Reply:
I think having it configurable via config makes sense. We do that for db role rotations. In that case, we verify that we don't fall below a lower bound of 5 seconds. I don't think we have an upper bound. Maybe we don't need an upper bound here?

Reply:
I checked what we're doing in the database static roles and it looks like the only thing we check is, as JM mentioned, that we're above the tick rate for the PQ processing function, so I've removed the upper bound bit.

For the lower bound, I actually checked through the code and as far as I could find, the queue is being scanned as part of the Rollback periodic function attached to the framework/Backend, which I think is only run on HandleRequest? If there is no attached automatic periodic rotation check, I imagine we'd want to add one, like the `database` static roles version is doing? (Doing that is out of scope for this particular PR, I'd think)

Reply:
I'm going to merge this PR and defer discussion of whether we should add configurable limits to future discussion, since I _believe_ the current functionality [after changes] matches what we're doing in AWS.